### PR TITLE
race on `jsipfs init`

### PR DESCRIPTION
**Pull request by diasdavid**
sometimes, travis fails with a repo that wasn't fully initiated on`jsipfs init` (just CLI)

I'm being unable to replicate it locally, so this PR is just me throwing things to see if I can figure it out :)

### CODE REVIEW

**Comment by diasdavid** [comment]
Node.js stable passes on circle, but fails on travis.

File: src/http-api/index.js [hunk 1]
```diff
@@ -21,113 +21,115 @@ function uriToMultiaddr (uri) {
...
+            pubsub: true
+          }
+        })
 
-        this.server.start((err) => {
+        this.node.once('error', (err) => {
+          err.code = 'ENOENT'
+          cb(err)
+        })
+        this.node.once('start', cb)
```
Comment (dignifiedquire) [inline src/http-api/index.js]: need to wrap the `cb` into a  `once` otherwise there could be a double calling

File: src/http-api/index.js [hunk 2]
```diff
@@ -21,113 +21,115 @@ function uriToMultiaddr (uri) {
...
+
+          // Set default headers
+          setHeader(this.server,
+            'Access-Control-Allow-Headers',
+            'X-Stream-Output, X-Chunked-Output, X-Content-Length')
+          setHeader(this.server,
+            'Access-Control-Expose-Headers',
+            'X-Stream-Output, X-Chunked-Output, X-Content-Length')
+
+          this.server.start((err) => {
```
Comment (dignifiedquire) [inline src/http-api/index.js]: This can be just `this.server.start(cb)` and then the rest below be in the next block

File: src/http-api/index.js [hunk 3]
```diff
@@ -21,113 +21,115 @@ function uriToMultiaddr (uri) {
...
+            if (err) {
+              return callback(err)
```
Comment (dignifiedquire) [inline src/http-api/index.js]: this should be `cb`

File: src/http-api/index.js [hunk 4]
```diff
@@ -21,113 +21,115 @@ function uriToMultiaddr (uri) {
...
+            api.info.ma = uriToMultiaddr(api.info.uri)
+            gateway.info.ma = uriToMultiaddr(gateway.info.uri)
+
+            console.log('API is listening on: %s', api.info.ma)
+            console.log('Gateway (readonly) is listening on: %s', gateway.info.ma)
+
+            // for the CLI to know the where abouts of the API
+            fs.writeFileSync(apiFilePath, api.info.ma)
+
+            callback()
```
Comment (dignifiedquire) [inline src/http-api/index.js]: this should be `cb`

File: test/cli/daemon.js [hunk 5]
```diff
@@ -10,14 +10,15 @@ describe('daemon', () => {
   let ipfs
 
   beforeEach(() => {
-    repoPath = '/tmp/ipfs-test-' + Math.random().toString().substring(2, 8)
+    repoPath = '/tmp/ipfs-test-not-found-' + Math.random().toString().substring(2, 8)
```
Comment (dignifiedquire) [inline test/cli/daemon.js]: why `not-found`?

File: src/http-api/index.js (hunk 1 above)
Comment (diasdavid) [inline src/http-api/index.js]: but there is a `once` listener, so it only does it `once`

File: test/cli/daemon.js (hunk 5 above)
Comment (diasdavid) [inline test/cli/daemon.js]: I was just making it obvious for my tests :)

File: src/http-api/index.js (hunk 1 above)
Comment (dignifiedquire) [inline src/http-api/index.js]: but you register it twice, once for start and once for error

Comment (diasdavid) [inline src/http-api/index.js]: they are mutually exclusive, but I'll add it anyway just to be sure 👍

**Comment by diasdavid** [comment]
before I squashed, this was passing:
image

Now it is just hogged because travis is not even starting the tests. Going to go ahead and merge :)

**Pull request merged**